lib/handlers/subscribe.js

```diff
@@ -119,19 +125,30 @@ function addSubs (sub, done) {
   const broker = client.broker
   const topic = sub.topic
   const qos = sub.qos
-  let func = qos > 0 ? client.deliverQoS : client.deliver0
+  const rh = this.rh
+  const rap = this.rap
+  const nl = this.nl
+  const deliverFunc = qos > 0 ? client.deliverQoS : client.deliver0
+
+  let func = function handlePacketSubscription (_packet, cb) {
+    if (!rap) {
+      _packet = new Packet(_packet, broker)
+      _packet.retain = false
+    }
+    deliverFunc(_packet, cb)
+  }
```

Comment:
Please avoid to allocate this function if rap is set.

Reply:
Will do.